src/rest/RequestHandler.js

```diff
@@ -125,6 +143,20 @@ class RequestHandler {
         });
       }
 
+      // Determine whether a RateLimitError should be thrown
+      await this.onRateLimit(request, limit, timeout, isGlobal); // eslint-disable-line no-await-in-loop
```

Comment:
Here's a small issue. If we throw on a global ratelimit, before preventing every other request from going, we could hit lots of 429s real fast. This should be moved before the await to the delayPromise 👍

Reply:
i'm not sure if i understood correctly, you mean that line 153 (`this.manager.globalDelay = this.globalDelayFor(timeout);`) should go before this line?

Reply:
No, these two lines should go before current line 160 and 161

Reply:
yeah thats what i meant. Good catch